src/librustc/session/config.rs

```diff
@@ -515,6 +515,8 @@ options! {CodegenOptions, CodegenSetter, basic_codegen_options,
         "explicitly enable the cfg(debug_assertions) directive"),
     inline_threshold: Option<usize> = (None, parse_opt_uint,
         "set the inlining threshold for"),
+    disable_gold: bool = (false, parse_bool,
```

Comment:
This is… a pretty confusing option (both its description and behaviour).

We already have `-C linker` which possibly swaps out `gcc` for something else and now we pass a special flag into linker assuming it is `gcc`-compatible (in absence of this option) without checking if specified linker is indeed compatible. That might make `-C linker` harder to use.

Maybe it would be better to have `-C linker=gcc` which overrides any default/dubious flags we might add?